homeassistant/components/shelly/sensor.py

```diff
@@ -1082,6 +1082,9 @@ class RestSensorDescription(RestEntityDescription, SensorEntityDescription):
             or config[key]["enable"] is False
             or status[key].get("xpercent") is None
         ),
+        unit=lambda config: config["xpercent"]["unit"]
+        if config["xpercent"]["unit"]
+        else None,
```

Comment:
```suggestion
        unit=lambda config: config["xpercent"]["unit"] or None
```

Reply:
If unit is never `""` you could drop the `or None` as well

Reply:
The documentation says that the `unit` value can be `null` or a `[0..20]` chars string, so I'm afraid that an empty string might happen

![obraz](https://github.com/user-attachments/assets/dc4f32df-5f06-40ab-aaa0-ca791d956675)

Reply:
I'll ask Shelly team about this, maybe we'll have a clearer situation.

Reply:
I will check what is possible to set with a real device and report back

Reply:
I tried to use the API and the UI to set values for unit, when an empty string is set it will return `null` when read back. I think it is safe to use `unit=lambda config: config["xpercent"]["unit"] `

Reply:
Thanks guys!